UI decoration

myogh opened the pull request:
I've made some style changes to the grids. It seems to work fine with the **show animation** option enabled. 
When it's disabled, errors pop up regarding threads.

Commit:
Add margin bwt grids

Commit:
Add dropshadow and vivid color in visualization

mlsdpk commented on src/grid_map.cpp:
```diff
@@ -102,11 +102,23 @@ void GridMap::renderGrids(int grid_size) {
     for (int row = 0; row < no_of_row_grids_; row++) {
+
+      // create shadoweffect obj
+      QGraphicsDropShadowEffect *e = new QGraphicsDropShadowEffect;
```
I believe these shadow effects and graphic stuffs inside this loop are for visualization of initial grid (which is free space). I think it's better to put these inside actual grid object, may b in constructor of Grid?

myogh replied:
I have made this change.

mlsdpk commented on src/mainwindow.cpp:
```diff
@@ -219,17 +219,33 @@ void MainWindow::receiveChildParentIdxs(
         gridMap->getGridsList()->at(idx)->setBrush(
-            QBrush(Qt::blue, Qt::SolidPattern));
-    }
+            QBrush(QColor(200, 200, 200, 245), Qt::SolidPattern));
```
Same goes to here too. How about we create some setter functions in Grid for creating visuals and call these inside this loop?

mlsdpk replied:
And I think thread issue happens since this function runs in different therad, you can fix this by removing `Qt::DirectConnection` in line 343.

mlsdpk commented on src/mainwindow.cpp:
```diff
@@ -219,17 +219,33 @@ void MainWindow::receiveChildParentIdxs(
       // find the parent vertex of current key (child)
       currentPathKeyIdx_ = childParentIdxs_[currentPathKeyIdx_];
+      // grid styling
+      gridMap->getGridsList()
```
We can create one setter for path visualization also.

mlsdpk commented on src/planner/search/dfs/dfs.cpp:
```diff
@@ -1,6 +1,7 @@
 #include "planner/search/dfs/dfs.h"
 
 #include <iostream>
+#include <cmath>
```
We can remove this if we don't need.

myogh replied:
In the building process, the compiler says `std::sqrt` is invalid. That's why I had to add **cmath**.

mlsdpk reviewed (changes requested):
Thank you! Looks good to me, i've suggested few changes before we merge it.

Commit:
Moved grid initial styles to its constructor

Commit:
Removed shadow effect in no animaition mode

- Change opacity of the grids upon showing path
- Unable to setBrush grids in no animation mode due to
QGraphicsDropShadow

myogh commented:
I added this new [commit](3d13d83b0e3d648b4aa8887dc0e0b6626d1bacda). 
Either brushing the Grid nor adding dropshadow was failed to do in no animation mode. I think it's happening because of the `QGraphicsDropShadowEffect`.

If drop shadow effect can be elimented in no animation mode as it has already been done, the ui part is mostly done. Everything is working perfectly now.

mlsdpk committed:
Refactor graphic effects

mlsdpk commented:
I've updated and moved the visualization stuffs into grid class. If you're ok with current one, we can proceed to merge with the main branch. Please let me know after you've tested @aungmcs.

myogh commented:
@mlsdpk, I've tested it and there're no issues at all. Merging can be done without any hassle.

mlsdpk commented:
Great! Thank you for your contribution. I'm closing this PR for now.
